### PR TITLE
Fix numpy warnings

### PR DESCRIPTION
**Pull request by dabacon**
Fixes a bunch of numpy warnings.  These were all about `np.bool` and `np.complex` and `np.int` being deprecated as they are just direct aliases of `bool`, `complex`, and `int`.

One interesting case is json test where I kept the actual numpy scalar type using `np.bool_`.

### CODE REVIEW

**Comment by dabacon** [comment]
Fixes #3975

File: cirq-core/cirq/protocols/json_serialization_test.py [hunk 1]
```diff
@@ -232,7 +232,7 @@ def test_builtins():
 def test_numpy():
     x = np.ones(1)[0]
 
-    assert_json_roundtrip_works(x.astype(np.bool))
+    assert_json_roundtrip_works(x.astype(np.bool_))
```
Comment (balopat) [inline cirq-core/cirq/protocols/json_serialization_test.py:235]: maybe add a comment why this is interesting? It seems like a private type.

**Review by balopat — APPROVED** [review]
LGTM with a request for a comment on `np.bool_`

File: cirq-core/cirq/protocols/json_serialization_test.py (hunk 1 above)
Comment (dabacon) [reply cirq-core/cirq/protocols/json_serialization_test.py]: I read more, and I don't think we need this check, the objects we will get will al be dtype bool, so we should be same.

**Pull request merged**